### ISSUE TITLE
Potential improvements to HandleSuccessRequestAsync

### ISSUE THREAD

**Issue by sondreb** [proudmonkey/AutoWrapper#5]
The HandleSuccessRequestAsync method will attempt to perform a check for valid json, convert to JSON string and then serialize to JSON, the response from a call. This happens even if the content being served is HTML. This is highly inefficient and there is probably a much better way to ignore the JSON deserialization when the code discovers the content is not JSON.

![image](https://user-images.githubusercontent.com/309938/66145403-1140d100-e60b-11e9-9e18-bc4a2556ebc6.png)

**Comment by sondreb** [comment]
As most APIs is located under /api, I decided to copy the AutoWrapper source code and add this extra guard check in the invoke method:

![image](https://user-images.githubusercontent.com/309938/66146354-e788a980-e60c-11e9-8113-0766396657bb.png)

This also fixes issue #4 

If the API is under another path, that could be configured with AutoWrapperOptions. It is an issue though if you're hosting ApiControllers on root, like the default controller with a new ASP.NET web app where "WeatherForecastController" is accessible through /weatherforecast.

**Comment by proudmonkey** [comment]
That's interesting! I haven't tested that scenario.  Care to share what you've done inside the `IsApi()` method?

 I will revisit the code. I agree that non JSON format should be ignored.

Thank you so much for your feedback!

**Comment by sondreb** [comment]
IsApi is just a copy of IsSwagger:

```C#
        private bool IsApi(HttpContext context)
        {
            return context.Request.Path.StartsWithSegments("/api");
        }
```

**Comment by proudmonkey** [comment]
I've released version `1.1.0` and you should now be able to set `IsApiOnly` in the `AutoWrapperOptions`. For example:

```
   app.UseApiResponseAndExceptionWrapper( new AutoWrapperOptions { IsApiOnly = false });
```

You can also use the `WrapWhenApiPathStartsWith` option to set your Api path to validate. For example:

```
app.UseApiResponseAndExceptionWrapper( new AutoWrapperOptions { IsApiOnly = false, WrapWhenApiPathStartsWith = "/myapi" });
```

I would still recommend you to create your APIs in a separate project so you won't confuse yourself with Angular and ASP.NET Core API routings.
